sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -56,6 +64,33 @@ trait Predicate extends Expression {
   override def dataType: DataType = BooleanType
 }
 
+/**
+ * The factory object for `BasePredicate`.
+ */
+object Predicate extends CodeGeneratorWithInterpretedFallback[Expression, BasePredicate] {
+
+  override protected def createCodeGeneratedObject(in: Expression): BasePredicate = {
+    GeneratePredicate.generate(in)
+  }
+
+  override protected def createInterpretedObject(in: Expression): BasePredicate = {
+    InterpretedPredicate(in)
+  }
+
+  /**
+   * Returns a BasePredicate for a bound Expression.
+   */
+  def create(expr: Expression): BasePredicate = {
+    create(expr)
+  }
+
```

Comment:
We can add a method `def createInterpreted...` for places that want to use interpreted predicates.

Reply:
Yea. ok.